drivers/counter/counter_handlers.c

```diff
@@ -11,38 +11,138 @@
-#include <syscalls/counter_read_mrsh.c>
 #include <syscalls/counter_stop_mrsh.c>
 #include <syscalls/counter_start_mrsh.c>
+
+static inline bool z_vrfy_counter_is_counting_up(const struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_OBJ(dev, K_OBJ_DRIVER_COUNTER));
+	return z_impl_counter_is_counting_up((const struct device *)dev);
+}
+#include <syscalls/counter_is_counting_up_mrsh.c>
+
+static inline u8_t z_vrfy_counter_get_num_of_channels(const struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_OBJ(dev, K_OBJ_DRIVER_COUNTER));
+	return z_impl_counter_get_num_of_channels((const struct device *)dev);
+}
+#include <syscalls/counter_get_num_of_channels_mrsh.c>
+
+static inline u32_t z_vrfy_counter_get_frequency(const struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_OBJ(dev, K_OBJ_DRIVER_COUNTER));
+	return z_impl_counter_get_frequency((const struct device *)dev);
+}
+#include <syscalls/counter_get_frequency_mrsh.c>
+
+static inline u32_t z_vrfy_counter_us_to_ticks(const struct device *dev,
+					       u64_t us)
+{
+	Z_OOPS(Z_SYSCALL_OBJ(dev, K_OBJ_DRIVER_COUNTER));
+	return z_impl_counter_us_to_ticks((const struct device *)dev,
+					  (u64_t)us);
+}
+#include <syscalls/counter_us_to_ticks_mrsh.c>
+
+static inline u64_t z_vrfy_counter_ticks_to_us(const struct device *dev,
+					       u32_t ticks)
+{
+	Z_OOPS(Z_SYSCALL_OBJ(dev, K_OBJ_DRIVER_COUNTER));
+	return z_impl_counter_ticks_to_us((const struct device *)dev,
+					  (u32_t)ticks);
+}
+#include <syscalls/counter_ticks_to_us_mrsh.c>
+
+static inline u32_t z_vrfy_counter_read(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_COUNTER(dev, read));
+	return z_impl_counter_read((struct device *)dev);
+}
+#include <syscalls/counter_read_mrsh.c>
+
+static inline int z_vrfy_counter_set_channel_alarm(struct device *dev,
+			u8_t chan_id, const struct counter_alarm_cfg *alarm_cfg)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_COUNTER(dev, set_alarm));
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((const struct counter_alarm_cfg *)
+				      alarm_cfg,
+				      sizeof(struct counter_alarm_cfg)));
+	Z_OOPS(Z_SYSCALL_VERIFY_MSG(alarm_cfg->callback == 0,
+				    "callbacks may not be set from user mode"));
+	return z_impl_counter_set_channel_alarm((struct device *)dev,
+		(u8_t)chan_id, (const struct counter_alarm_cfg *)alarm_cfg);
+
+}
+#include <syscalls/counter_set_channel_alarm_mrsh.c>
+
+static inline int z_vrfy_counter_cancel_channel_alarm(struct device *dev,
+						      u8_t chan_id)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_COUNTER(dev, cancel_alarm));
+	return z_vrfy_counter_cancel_channel_alarm((struct device *)dev,
+						   (u8_t)chan_id);
+}
+#include <syscalls/counter_cancel_channel_alarm_mrsh.c>
+
+static inline int z_vrfy_counter_set_top_value(struct device *dev,
+					       const struct counter_top_cfg
+					       *cfg)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_COUNTER(dev, set_top_value));
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((const struct counter_top_cfg *)cfg,
+				      sizeof(struct counter_top_cfg)));
+	Z_OOPS(Z_SYSCALL_VERIFY_MSG(cfg->callback == 0,
```

Comment:
see above note about copying the config struct

Reply:
@andrewboie Thank you. Fixed.